src/Cardano/AddressDerivation.hs

```diff
@@ -202,3 +223,48 @@ class GenMasterKey (key :: Depth -> * -> *) where
 
     -- | Generate a root key from a corresponding seed.
     genMasterKey :: GenMasterKeyFrom key -> ScrubbedBytes -> key 'RootK XPrv
+
+
+-- | Address is mere wrapper around a ByteString and represents an encoded address.
+newtype Address = Address
+    { unAddress :: ByteString
+    } deriving (Show, Generic, Eq, Ord)
+
+instance NFData Address
+
+-- | Available network options.
+data NetworkDiscriminant = Mainnet | Testnet Nat
```

Comment:
From this module perspective though, the fact that something is mainnet or testnet is anecdotal. What really matters is whether addresses requires magic or don't. 

I'd suggest to reflect this as `RequiresMagic | RequiresNoMagic`.